### PR TITLE
Purchases: Always use selectedSite.slug in urls when selected site needs to be used

### PR DESCRIPTION
**Pull request by gziolo**
This PR addresses [comment](https://github.com/Automattic/wp-calypso/pull/1036/files#r46307589) from @drewblaisdell:
> We should always be using whatever `SitesList` set to `slug` here, which is usually the domain, but switches to a `[site].wordpress.com` if there are conflicts.

It also tries to use the same style of invoking `managePurchase` url helper where possible.

### Testing
This PR affects the links on every page under `/purchases`, including the following components:

`CancelPrivateRegistration`
`CancelPurchase`
`CancelPurchaseLoadingPlaceholder`
`CancelPurchaseProductInformation`
`ConfirmCancelPurchase`
`ManagePurchase`
`EditCardDetails`

We can test this by asserting that the links from every page under `/purchases` to another page in `/purchases` are not broken, and use the site slug (either `foobar.wordpress.com` or `foobar.com`, depending on if the site has a primary domain or if there are namespace conflicts) in the URL for each link.

### Review
* [x] QA review
* [x] Code review

### CODE REVIEW

File: client/me/purchases/manage-purchase/index.jsx [hunk 1]
```diff
@@ -414,7 +414,7 @@ const ManagePurchase = React.createClass( {
 
 		if ( isIncludedWithPlan( purchase ) ) {
 			const attachedPlanUrl = paths.managePurchase(
-				this.props.selectedSite.domain,
+				this.props.domain,
```
Comment (drewblaisdell) [inline client/me/purchases/manage-purchase/index.jsx]: `this.props.domain` isn't defined here. I believe should be `this.props.selectedSite.slug`.

There is a bug here though: this block will never render if the purchase is expiring. I'll add a commit to address both of these issues.

**Comment by drewblaisdell** [comment]
Thanks for opening this, @gziolo. I found a few more cases that needed updating. I added a single commit for the changes in each component so that this is easier to review, but we should squash them before merging.

As far as testing, we need to assert that links to other `/purchases` routes still work in the following components:

`CancelPrivateRegistration`
`CancelPurchase`
`CancelPurchaseLoadingPlaceholder`
`CancelPurchaseProductInformation`
`ConfirmCancelPurchase`
`ManagePurchase`
`EditCardDetails`

Of note while testing: until #536 is merged, `PurchasesList` will still link to `ManagePurchase` with `domain` instead of `slug`.

**Comment by gziolo** [comment]
@drewblaisdell thank you for your commits. Let's merge #536 and revisit this one to make sure all places are properly updated. At the moment this one is hard to test because of dependencies on mentioned PR.

Looks good so far and that should be it :+1:

**Comment by fabianapsimoes** [comment]
Moving this back to `In Progress` until #536 is merged.

**Comment by drewblaisdell** [comment]
I added another commit to update `PurchasesUtils`, but the testing instructions shouldn't have changed. I also rebased.

#536 is merged, so this should be back in review.

**Comment by gziolo** [comment]
I rebase with master and added tiny refactoring which takes advantage of `getSelectedSite` introduced in another PR. Code looks good to merge :+1:

**Comment by gziolo** [comment]
I found hopefully last missing url to update. It was `Cancel and Refund` button for purchase within refund period. Now it should load `Confirm Cancel Purchase` page with proper slug in url. All other uses cases worked properly for me :+1:

**Comment by drewblaisdell** [comment]
@gziolo :+1:

I squashed a few of my commits. This is ready to merge now.

**Comment by gziolo** [comment]
@drewblaisdell nice, thanks for help on this one :)

**Pull request merged**